cluster/service/cluster_service_blackbox_test.go

```diff
@@ -533,6 +533,138 @@ func (s *ClusterServiceTestSuite) TestLoadForAuth() {
 		})
 	})
 }
+func (s *ClusterServiceTestSuite) TestFindByURL() {
+
+	s.T().Run("ok", func(t *testing.T) {
+
+		for _, saName := range []string{"fabric8-oso-proxy", "fabric8-tenant", "fabric8-jenkins-idler", "fabric8-jenkins-proxy", "fabric8-auth"} {
+			t.Run(saName, func(t *testing.T) {
+				// given
+				c := newTestCluster()
+				err := s.Application.ClusterService().CreateOrSaveCluster(context.Background(), c)
+				require.NoError(t, err)
+				sa := &authtestsupport.Identity{
+					Username: saName,
+					ID:       uuid.NewV4(),
+				}
+				ctx, err := authtestsupport.EmbedServiceAccountTokenInContext(context.Background(), sa)
+				require.NoError(t, err)
+				// when
+				result, err := s.Application.ClusterService().FindByURL(ctx, c.URL)
+				// then
+				require.NoError(t, err)
+				require.NotNil(t, result)
+				test.AssertEqualCluster(t, *c, *result, false)
+			})
+		}
+	})
+
+	s.T().Run("failures", func(t *testing.T) {
+
+		t.Run("unauthorized", func(t *testing.T) {
+			// given
+			c := newTestCluster()
+			err := s.Application.ClusterService().CreateOrSaveCluster(context.Background(), c)
+			require.NoError(t, err)
+			sa := &authtestsupport.Identity{
+				Username: "foo",
+				ID:       uuid.NewV4(),
+			}
+			ctx, err := authtestsupport.EmbedServiceAccountTokenInContext(context.Background(), sa)
+			require.NoError(t, err)
+			// when
+			_, err = s.Application.ClusterService().FindByURL(ctx, c.URL)
+			// then
+			testsupport.AssertError(t, err, errors.UnauthorizedError{}, "unauthorized access to cluster info")
+		})
+
+		t.Run("not found", func(t *testing.T) {
+			// given
+			c := newTestCluster()
+			err := s.Application.ClusterService().CreateOrSaveCluster(context.Background(), c)
+			require.NoError(t, err)
+			sa := &authtestsupport.Identity{
+				Username: "fabric8-auth",
+				ID:       uuid.NewV4(),
+			}
+			ctx, err := authtestsupport.EmbedServiceAccountTokenInContext(context.Background(), sa)
+			require.NoError(t, err)
+			// when
+			_, err = s.Application.ClusterService().FindByURL(ctx, "http://foo")
+			// then
+			require.Error(t, err)
+			testsupport.AssertError(t, err, errors.NotFoundError{}, fmt.Sprintf("cluster with url '%s' not found", "http://foo"))
+		})
+	})
+}
+
+func (s *ClusterServiceTestSuite) TestFindByURLForAuth() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		for _, saName := range []string{"fabric8-auth"} {
+			t.Run(saName, func(t *testing.T) {
+				// given
+				c := newTestCluster()
+				err := s.Application.ClusterService().CreateOrSaveCluster(context.Background(), c)
+				require.NoError(t, err)
+				sa := &authtestsupport.Identity{
+					Username: saName,
+					ID:       uuid.NewV4(),
+				}
+				ctx, err := authtestsupport.EmbedServiceAccountTokenInContext(context.Background(), sa)
+				require.NoError(t, err)
+				// when
+				result, err := s.Application.ClusterService().FindByURLForAuth(ctx, c.URL)
```

Comment:
Same as above (trailing slashes and without it)

Reply:
done in e4586bc